# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1460471796 14400
#      Tue Apr 12 10:36:36 2016 -0400
# Node ID 9bb39b754ab131fb3fec1a68b12bc4661f282c3d
# Parent  e15d3387fc595a05b9d9845fe594b5e9292ea26f
finish previous change to octave_env::set_program_name

* oct-env.cc (octave_env::set_program_name):
Only perform initialization once.

diff --git a/liboctave/system/oct-env.cc b/liboctave/system/oct-env.cc
--- a/liboctave/system/oct-env.cc
+++ b/liboctave/system/oct-env.cc
@@ -256,17 +256,17 @@ octave_env::chdir (const std::string& ne
 {
   return (instance_ok ())
          ? instance->do_chdir (newdir) : false;
 }
 
 void
 octave_env::do_set_program_name (const std::string& s) const
 {
-  bool initialized = false;
+  static bool initialized = false;
 
   if (! initialized)
     {
       // The string passed to gnulib's ::set_program_name function must
       // exist for the duration of the program so allocate a copy here
       // instead of passing S.c_str () which only exists as long as the
       // string object S.
 
@@ -278,16 +278,18 @@ octave_env::do_set_program_name (const s
 
       size_t pos
         = prog_invocation_name.find_last_of (file_ops::dir_sep_chars ());
 
       // Also keep a shortened version of the program name.
       prog_name = (pos == std::string::npos
                    ? prog_invocation_name
                    : prog_invocation_name.substr (pos+1));
+
+      initialized = true;
     }
 }
 
 // Return a pretty pathname.  If the first part of the pathname is the
 // same as $HOME, then replace that with '~'.
 
 std::string
 octave_env::do_polite_directory_format (const std::string& name) const
